File: dotnet-desktop-guide/net/winforms/controls/snippets/how-to-make-thread-safe-calls/vb/FormBackgroundWorker.vb
Public Class FormBackgroundWorker
    '<Background>
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click

        If (Not BackgroundWorker1.IsBusy) Then
            BackgroundWorker1.RunWorkerAsync()
        End If

    End Sub

    Private Sub BackgroundWorker1_DoWork(sender As Object, e As ComponentModel.DoWorkEventArgs) Handles BackgroundWorker1.DoWork

        Dim counter = 0
        Dim max = 10

        While counter <= max

            BackgroundWorker1.ReportProgress(0, counter.ToString())
            System.Threading.Thread.Sleep(1000)

            counter += 1

        End While

    End Sub

    Private Sub BackgroundWorker1_ProgressChanged(sender As Object, e As ComponentModel.ProgressChangedEventArgs) Handles BackgroundWorker1.ProgressChanged
        TextBox1.Text = e.UserState
    End Sub
    '</Background>
End Class

File: dotnet-desktop-guide/winforms/controls/snippets/how-to-make-thread-safe-calls/cs/FormBackgroundWorker.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace project
{
    public partial class FormBackgroundWorker : Form
    {
        public FormBackgroundWorker()
        {
            InitializeComponent();
        }

        //<Background>
        private void button1_Click(object sender, EventArgs e)
        {
            if (!backgroundWorker1.IsBusy)
                backgroundWorker1.RunWorkerAsync();
        }

        private void backgroundWorker1_DoWork(object sender, DoWorkEventArgs e)
        {
            int counter = 0;
            int max = 10;

            while (counter <= max)
            {
                backgroundWorker1.ReportProgress(0, counter.ToString());
                System.Threading.Thread.Sleep(1000);
                counter++;
            }
        }

        private void backgroundWorker1_ProgressChanged(object sender, ProgressChangedEventArgs e) =>
            textBox1.Text = (string)e.UserState;
        //</Background>
    }
}
